File: EatRegerator/Services/Eat/Classes/Eat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EatRegerator.Services.Classes
{
  public class Dishes
  {
    public Guid DishGuid { get; set; }
    public int? CookingTime { get; set; }
    public Guid TypeGuid { get; set; }
    public string Title { get; set; }
    public Guid TypeKitchenGuid { get; set; }
    public Guid TypeMenuGuid { get; set; }
    public string PictureUrl { get; set; }
    public string Description { get; set; }
  }

  public class Products
  {
    public Guid ProductGuid { get; set; }
    public string Name { get; set; }
  }

  public class Recipes
  {
    public Guid RecipeGuid { get; set; }
    public string Text { get; set; }
    public int Order { get; set; }
    public string PictureUrl { get; set; }
    public string Title { get; set; }
    public Guid DishGuid { get; set; }
  }

  public class TypeDishes
  {
    public Guid TypeGuid { get; set; }
    public string Title { get; set; }
    public int Code { get; set; }
  }

  public class TypeKitchen
  {
    public Guid KitchenTypeGuid { get; set; }
    public string Title { get; set; }
    public int Code { get; set; }
  }

  public class TypeMenu
  {
    public Guid TypeMenuGuid { get; set; }
    public string Title { get; set; }
    public int Code { get; set; }
  }

  public class ProductsResult : BaseResult
  {
    public List<Products> Products { get; set; }
  }

  public class TypeDishesResult : BaseResult
  {
    public List<TypeDishes> TypeDishes { get; set; }
  }

  public class TypesKitchensResult : BaseResult
  {
    public List<TypeKitchen> TypesKitchen { get; set; }
  }

  public class TypesMenuResult : BaseResult
  {
    public List<TypeMenu> TypesMenu { get; set; }
  }

  public class DishesResult : BaseResult
  {
    public List<Dishes> Dishes { get; set; }
  }

  public class RecipeResult : BaseResult
  {
    public DishRecipe Recipe { get; set; }
  }

  public class DishRecipe : Dishes
  {
    public List<Recipes> Recipe { get; set; }
  }

  public class GetDishesInput
  {
    public List<Guid> IncreaseProductGuids { get; set; }
    public List<Guid> DecreaseProductGuids { get; set; }
    public Guid? TypeDishesGuid { get; set; }
    public Guid? TypeKitchensGuid { get; set; }
    public Guid? TypeMenuGuid { get; set; }
  }
}

File: EatRegerator/ClientApp/src/services/eat-service/eat.ts
import { BaseResponse } from "../BaseResponse";

export class ProductsResult extends BaseResponse {
  public products: Array<Products>
}

export class TypeDishesResult extends BaseResponse {
  public typeDishes: Array<TypeDishes>
}

export class TypesKitchensResult extends BaseResponse {
  public typesKitchen: Array<TypeKitchen>
}

export class TypesMenuResult extends BaseResponse {
  public typesMenu: Array<TypeMenu>
}

export class DishesResult extends BaseResponse {
  public dishes: Array<Dishes>;
}

export class Dishes {
  public dishGuid: string;
  public cookingTime: number;
  public typeGuid: string;
  public title: string;
  public typeKitchenGuid: string;
  public typeMenuGuid: string;
  public pictureUrl: string;
  public description: string;
}

export class Products {
  public productGuid: string;
  public name: string;
}

export class TypeDishes {
  public typeGuid: string;
  public title: string;
  public code: number;
}


export class TypeKitchen {
  public kitchenTypeGuid: string;
  public title: string;
  public code: number;
}

export class TypeMenu {
  public typeMenuGuid: string;
  public title: string;
  public code: number;
}

export class GetDishesInput {
  constructor() {
    this.increaseProductGuids = [];
    this.decreaseProductGuids = [];
  }
  public increaseProductGuids: Array<string>;
  public decreaseProductGuids: Array<string>;
  public typeDishesGuid: string;
  public typeKitchensGuid: string;
  public typeMenuGuid: string;
}

export class Recipes {
  public recipeGuid: string;
  public text: string;
  public order: number;
  public pictureUrl: string;
  public title: string;
}

export class RecipeResult extends BaseResponse
{
  public recipe: DishRecipe;
}

export class DishRecipe extends Dishes
{
  public recipe: Array<Recipes>;
}
